src/main/java/com/timeplus/kafkaconnect/TimeplusSinkTask.java

```diff
@@ -90,21 +98,41 @@ public void put(Collection<SinkRecord> records) {
                 .post(body)
                 .build();
 
+        client.interceptors().add(new Interceptor() {
+            @Override
+            public Response intercept(Chain chain) throws IOException {
+                Request request = chain.request();
+
+                // try the request
+                Response response = chain.proceed(request);
+                int tryCount = 0;
+                while ((response.code() == 429 || response.code() >= 500) && tryCount < MAX_RETRY_COUNT) {
```

Comment:
Shall we just use an existing java lib for retry? e.g. https://failsafe.dev/

Reply:
I can use, but I not sure which lib is better to use, any recommendation?

Reply:
using failsafe now